fs/asyncreader/asyncreader.go

```diff
@@ -180,6 +180,77 @@ func (a *AsyncReader) WriteTo(w io.Writer) (n int64, err error) {
 	}
 }
 
+// SkipBytes will try to seek 'skip' bytes relative to the current position.
+// On success it returns true. If 'skip' is outside the current buffer data or
+// an error occurs, Abandon is called and false is returned.
+func (a *AsyncReader) SkipBytes(skip int) bool {
+	a.mu.Lock()
+	refillTokens := 0
+
+	if a.err != nil {
+		goto abandon
+	}
+	if skip < 0 {
+		// seek backwards if skip is inside current buffer
+		if a.cur != nil && a.cur.offset+skip >= 0 {
+			a.cur.offset += skip
+			a.mu.Unlock()
+			return true
+		}
+		goto abandon
+	}
+	// early return if skip is past the maximum buffer capacity
+	if skip >= a.buffers*BufferSize {
```

Comment:
Might this be better as this to only skip data which has been read?

```go
if skip >= len(a.ready)*BufferSize {
```

Or maybe that should be`len(a.ready)+1` - not sure.

Reply:
Good point, `len(a.ready)+1` is closer to the real buffer length than `a.buffers`.